### PR TITLE
arm64: enable acpi for qemu/virt.

### PR DESCRIPTION
**Pull request by jongwu**
acpi is enabled for kata 1.x, port and rebase code for 2.x
including:
runtime: enable pflash;
agent: add acpi support for pci bus path;
packaging: enable CONFIG_RTC_DRV_EFI;

Fixes: #1317
Signed-off-by: Jianyong Wu <jianyong.wu@arm.com>

@bergwolf  @jodh-intel

### CODE REVIEW

**Comment by jongwu** [comment]
hello @jodh-intel @devimc , any comments?

File: src/runtime/pkg/katautils/config.go [hunk 1]
```diff
@@ -228,6 +229,23 @@ func (h hypervisor) firmware() (string, error) {
 	return ResolvePath(p)
 }
 
+func (h hypervisor) PFlash() ([]string, error) {
+	pflashes := h.PFlashList
+
+	if len(pflashes) == 0 {
+		return []string(nil), nil
```
Comment (devimc) [inline src/runtime/pkg/katautils/config.go:236]: I know that `[]string(nil)` and `[]string{}` are the same, but can you use  `[]string{}` as the rest of functions that return a `[]string` ?

File: src/runtime/pkg/katautils/config.go [hunk 2]
```diff
@@ -228,6 +229,23 @@ func (h hypervisor) firmware() (string, error) {
...
+	}
+
+	for _, pflash := range pflashes {
+		_, err := ResolvePath(pflash)
+		if err != nil {
+			return []string{}, fmt.Errorf("pflash path doesn't exist: %s", pflash)
```
Comment (devimc) [inline src/runtime/pkg/katautils/config.go:242]: please include `err` in the message

**Review by devimc — COMMENTED** [review]
thanks @jongwu - I've left some comments

File: src/runtime/pkg/katautils/config.go (hunk 2 above)
Comment (jongwu) [reply src/runtime/pkg/katautils/config.go:242]: thanks @devimc

File: src/runtime/pkg/katautils/config.go (hunk 1 above)
Comment (jongwu) [reply src/runtime/pkg/katautils/config.go:236]: OK

**Review by devimc — APPROVED** [review]
thanks @jongwu

**Review by jodh-intel — APPROVED** [review]
Thanks @jongwu.

lgtm

**Comment by Jakob-Naucke** [comment]
@jodh-intel Thanks for the heads-up! Will look into it. Quickly judging by `grep -r create_pci_root_bus_path`, the only use atm is for [blockdev uevents](https://github.com/kata-containers/kata-containers/blob/main/src/agent/src/uevent.rs#L54) i.e. volumes (?) i.e. Kubernetes instead of just `ctr`/`crictl` (?). Since this is already upstream, I don't see a blocker for this PR though.

**Pull request merged**